yt_dlp/extractor/lbry.py

```diff
@@ -342,11 +348,27 @@ def _fetch_page(self, claim_id, url, params, page):
                 'url': self._permanent_url(url, stream_claim_name, stream_claim_id),
             }
 
+    def _fetch_playlist_items(self, claim_id):
+        page_params = {
+            'claim_ids': [claim_id],
+            'no_totals': True,
+            'page': 1,
+            'page_size': self._PAGE_SIZE,
+        }
+        return self._call_api_proxy('claim_search', claim_id, page_params, 'playlist')
+
     def _real_extract(self, url):
-        display_id = self._match_id(url).replace(':', '#')
-        result = self._resolve_url(
-            'lbry://' + display_id, display_id, 'channel')
-        claim_id = result['claim_id']
+        if playlist_id := self._match_valid_url(url).group('id2'):
+            is_playlist = True
+            r = self._fetch_playlist_items(playlist_id)
+            result = r.get('items', [])[0]
+            claim_ids = result.get('value', {}).get('claims', [])
```

Comment:
I would inline this function and use `traverse_obj`:
```suggestion
            claim_ids = traverse_obj(
                self._call_api_proxy('claim_search', playlist_id, {
                    'claim_ids': [playlist_id],
                    'no_totals': True,
                    'page': 1,
                    'page_size': self._PAGE_SIZE,
                }, 'playlist'), ('items', 0, 'value', 'claims', ...))
```
Additionally, if there could be more items in the result you could use `('items', ..., 'value', 'claims', ...)` to get all claims from all items.

Reply:
If went with item 0 for now (if a playlist conveying more than 50 items ever pop up, I could test it)
Also, I needed to maintain `result` to mimic the behavior similar to channels (`result` is used again at the end of the function)